mantidproject/mantid
# Total Scattering :: Add most basic form of total scattering workflow

* Added the most basic case for total scattering to the POLARIS instrument in the isis_powder diffraction scripts. 
* No real scientific meaning can be obtained from this case, but it should make a good foundation to build on moving forward (adding additional corrections + merging banks)
* To avoid continually updating a reference file, a system test has been added that measures peak intensity for the first peak of interest (~4 Angstrom).
* Updated POLARIS documentation with total scattering functionality

**To test:**
* Open Mantid plot
* Point your data search directory to: `\\olympic\babylon5\Public\ElliotOram\polaris` **(make sure this is the first data search directory in the list)**
* Also point the data search directory to the reference files directory `mantid\ExternalData\Testing\SystemTests\tests\analysis\reference`
* Run the following script
```
from isis_powder import Polaris
DIRS = config['datasearch.directories'].split(';')
# Relative to system data folder
working_folder_name = "polaris"


# Relative to input folder
calibration_folder_name = "calibration"
calibration_map_rel_path = "polaris_mapping.yaml"
config_rel_path = "Master_copy_of_grouping_file_with_essential_masks.cal"

# Generate paths
input_dir = os.path.join(DIRS[0], 'input')
output_dir = os.path.join(DIRS[0], 'output')

calibration_map_path = os.path.join(input_dir, calibration_map_rel_path)
calibration_dir = os.path.join(input_dir, calibration_folder_name)
config_path =os.path.join(calibration_dir, config_rel_path)

inst_obj = Polaris(user_name="Mantid-Test", calibration_mapping_file=calibration_map_path,
                   calibration_directory=calibration_dir, output_directory=output_dir, mode="PDF")

inst_obj.create_total_scattering_pdf(run_number='97947', merge_banks=False)
```
* A new group workspace (`pdf_output`) containing 5 workspaces should be present in the ADS
* These spectra should be Y unit: `PDF`,  X unit: `AtomicDistance`
* Plotting all banks together should look like this (zoomed in to avoid the majority of the initial tail):
![image](https://user-images.githubusercontent.com/13132128/33476046-6939fd86-d678-11e7-84b1-f02b7cc31bae.png)


* Try changing merge_banks to `True` (this should throw an exception as it is not yet implemented)

* Try loading in the focused data file to mantidplot and naming it either `97947-Results-TOF-Grp` or `97947-Results-D-Grp` then re-running the above script but changing the output directory to some other path (anywhere will do!) - this should ensure that the focused workspaces can be found using the ADS rather than only by file.
  * you could also do the ADS check with the output directory as shown in the script originally. In this case, ensure that no data is loaded and the workspaces come from the ADS rather than reloading the data - you should be able to assert this from looking at the result log.


Fixes #21294

**Release Notes** 
Under the total scattering section [here](https://github.com/mantidproject/mantid/blob/6c43079e50c9d0d36eb7a067388e6e7863360c0f/docs/source/release/v3.12.0/diffraction.rst)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
